query_panel/main.js

```diff
@@ -34,6 +34,75 @@ const app = createApp({
     };
   },
   methods: {
+    // Converts the provided data to CSV format.
+    dataToCsv(data) {
+      try {
+        if (!data || data.length === 0) {
+          console.error("No data available to convert to CSV");
+          return "";
+        }
+        const replacer = (key, value) => (value === null ? "" : value);
+        const header = Object.keys(data[0]);
+        const csv = [
+          header.join(","),
+          ...data.map((row) =>
+            header
+              .map((fieldName) => {
+                let fieldData = row[fieldName];
+                if (fieldData && typeof fieldData === "string") {
+                  fieldData = fieldData.replace(/"/g, '""'); // Escape double quotes
+                  return `"${fieldData}"`; // Wrap in double quotes
+                }
+                return JSON.stringify(fieldData, replacer);
+              })
+              .join(","),
+          ),
+        ].join("\r\n");
+        return csv;
+      } catch (error) {
+        console.error("Error converting data to CSV:", error);
+        return ""; // Return an empty string if there's an error
+      }
+    },
+    downloadAsCSV() {
+      const data = this.table.getData(); // Get the data the same way you do for copying
+      if (!data || data.length === 0) {
+        console.error("No data available for downloading.");
+        return;
+      }
+      let csvContent = this.dataToCsv(data);
```

Comment:
This method can return empty string, why?

Reply:
it returns empty in case of error

Reply:
I don't think exporting a empty file in case of an error is very useful. I would simply show an error message to the user that the file can't be generated with the error.